pandas/core/format.py

```diff
@@ -1895,6 +1895,17 @@ def get_formatted_cells(self):
 
 def format_array(values, formatter, float_format=None, na_rep='NaN',
                  digits=None, space=None, justify='right'):
+
+    if com.is_categorical_dtype(values):
+        values = values.get_values()
+
+    if isinstance(values, Index):
```

Comment:
I don't think you need 1902-1907. This is just checking dtype, its up to the formatter to format it. you want to delay converting to the lowest type here (and instead do it in the formatting part which is specific to that type).